README.md
# Weimer's Valen Mod Port for Enhanced Edition

By Hellmouse <hellmouse@outlook.fr>

## About

This mod is a port of the Weimer's Valen Mod (https://weidu.org/valen.html) for enhanced edition.
I did'nt touch the original content, only made it compatible enhanced edition and added an option to the installation for better EE and external mod compatibility(see compatiblity section below).
This version is only available on EE version, if you are using the Vanilla game, you would rather use the original mod https://weidu.org/valen.html

## Installation

An self extracting exe may be available later but for know, you need to manually copy & paste the file at the correct place

### Windows
- Download this repository from git and unzip the content
- Locate your game directory. It should be something like C:\Program Files\Beadmog\Baldur's Gate II Enhanced Edition or *path to your steam library*\SteamLibrary\steamapps\Baldur's Gate II Enhanced Edition
- Copy the "valen" folder of the archive into your game directory
- Copy the Setup-Valen.exe file into your game directory (not inside the valen folder !!!) your game directory should contain both the valen folder and the  Setup-Valen.exe file
- Run (click on) the  Setup-Valen.exe file
- Follow the instructions

### Mac & Linux

You only need to have your OS equivalent of the exe file. The exec file is actually the Weidu generic installer simply renamed for the mod name.

- Do the Windows steps until the copy of the Setup-Valen.exe one.
- Download the latest Weidu version for your OS on this website https://weidu.org/
- Copy the weidu file into your game directory and rename it Setup-Valen (left the extension unchanged)
- Run (click on) Setup-Valen the file 

## Compatibility


Most of the mod content has been tested and shoud run without issue. Non english translations, however have not been tested.

One of the original script made all the non evil and may cause incompatibility issues. In short, this component makes all the non evil priests hostile to the party if they see Valen. This can break the game (or at least a quest) if a non evil priest interaction with the party is required. I know there is at least one in Dorn quest, there may be many other in the EE content and / or other mods.

To avoid these conflicts, I made this component optional. You can choose:
- to include it and play the mod as Weimer intended to
- to remove it if it breaks your game (specially if you are using ohter mods that add quests/ NPC)

It would be possible to make this component less systematic and thus compatible with other mod, but this would be a modification of the *content* of the mod which is not the goal of this repository.



changelog.txt
Changelog of the enhanced port by Hellmouse

----
V0.0.1 (EE release)

This is the first github release, I imported the original Weimer's code into git and made the required updates via different commits. There are 5 types of change:

- no change at all, the file is exactly the same as Weimer's one, at the same path
- updated path: for easier compilation and/or code readability, the file has been moved to another directory but left unchanged
- recompile - the file has been uncompiled using weimer latest version to make sure it is compatible with the enhanced edition. The source has not been modified
- replaced with EE version the file has been replaced by an EE external file because it was not compatible at all.
- updated - the file content has been updated with the listed modifications


## unmodified files
- valen.cre (valen creature file)
- todo  (Weimer to do list)      
- valenend.2da  (2da file)
- README-Valen.txt readme file
- valen.spl  valenmst.spl

## moved only files (no change on the content)
- all .d files moved to /dialogs/ or /hunt1/dialogs/ for the hunter module
- all .tra folders moved to /languages
- all .itm (items) moved to /items

## recompiled file (no change)
- all scripts (.bcs files) -> moved to scripts/ or $componentName/scripts

## replace with EE version:
- Setup-Valen.exe replaced by latest weidu.exe
- portraits replaced by a photoshoped resized version (not ideal for quality, but I resizing is mandatory for the EE version and I did with what I had...)
- ids file replaced with an EE version (should be the lastest). This one could cause incompatibility bugs with mods that update the ids of the game

## changed
- Setup-Valen.tp2:
	- Update the path of moved files
	- Make the installer compiling the bcs file rather than copy compiled one
	- Make the priest component optional (in short, this component makes all the non evil priests hostile to the party if they see Valen. This can break the game (or at least a quest) if a non evil priest interaction with the party is required. I know there is at least one in Dorn quest, there may be many other in the EE content and / or other mods).

valen/dialogs/valentob.d
////////////////////////////////////////////////////////////////////////////
// Valen ToB Interjections
////////////////////////////////////////////////////////////////////////////

INTERJECT_COPY_TRANS ILLASERA 6 Valen_TOB0
  == "VALENJ"   IF ~InParty("Valen")~ THEN @1001
  == "ILLASERA" IF ~InParty("Valen")~ THEN @1002
END 

INTERJECT_COPY_TRANS SAREV25A 14 Valen_TOB1
  == "VALENJ"   IF ~InParty("Valen")~ THEN @1010
  == "SAREV25A" IF ~InParty("Valen")~ THEN @1011 = @1012
  == "VALENJ"   IF ~InParty("Valen")~ THEN @1013
  == "SAREV25A" IF ~InParty("Valen")~ THEN @1014
  == "VALENJ"   IF ~InParty("Valen")~ THEN @1015
  == "SAREV25A" IF ~InParty("Valen")~ THEN @1016
END 

INTERJECT_COPY_TRANS SAREV25A 55 Valen_TOB2
  == "VALENJ"   IF ~InParty("Valen")~ THEN @1020
  == "SAREV25A" IF ~InParty("Valen")~ THEN @1021
  == "VALENJ"   IF ~InParty("Valen")~ THEN @1022
  == "SAREV25A" IF ~InParty("Valen")~ THEN @1023
END

INTERJECT_COPY_TRANS SARMEL01 7 Valen_TOB3
  == "VALENJ"   IF ~InParty("Valen")~ THEN @1030 = @1031
  == "SARMEL01"   IF ~InParty("Valen")~ THEN @1032
END

// Saradush Jailer
EXTEND_BOTTOM SARJAI01 3
  IF ~InParty("Valen")~ THEN EXTERN VALENJ eat_jai
END

CHAIN3 VALENJ eat_jai
  @1033
  == "SARJAI01" @1034
  == "VALENJ" @1035
END SARJAI01 eat_jai2

APPEND SARJAI01
  IF ~~ THEN BEGIN eat_jai2
    SAY @1036
    IF ~~ THEN DO ~Enemy() ActionOverride("Valen",Attack("Sarjai01"))
      Unlock("Door03")~ EXIT
  END
END

INTERJECT SARPRO01 4 Valen_TOB4
  == "VALENJ"   IF ~InParty("Valen")~ THEN @1040
END SARPRO01 5

INTERJECT SARKIS01 9 Valen_TOB5
  == "VALENJ"   IF ~InParty("Valen")~ THEN @1050
END SARKIS01 16

INTERJECT SARMAG01 1 Valen_TOB6
  == "VALENJ"   IF ~InParty("Valen")~ THEN @1060
END SARMAG01 2

INTERJECT SARTHF2 0 Valen_TOB7
  == "VALENJ"   IF ~InParty("Valen") Global("Lazarus","GLOBAL",2)
                    Global("MuddyFootprints","GLOBAL",1)~ THEN @1070
  == "SARTHF2"   IF ~InParty("Valen") Global("Lazarus","GLOBAL",2)
                    Global("MuddyFootprints","GLOBAL",1)~ THEN @1071
  == "VALENJ"   IF ~InParty("Valen") Global("Lazarus","GLOBAL",2)
                    Global("MuddyFootprints","GLOBAL",1)~ THEN @1072
END SARTHF2 6

EXTEND_BOTTOM SARVOLO 9 #1
  IF ~InParty("Valen")~ THEN REPLY @1080 GOTO valen
END

CHAIN3 SARVOLO valen
  @1081 = @1082
  == "VALENJ" @1083
END SARVOLO 9

APPEND SARTEM01
  IF WEIGHT #-1 ~InParty("Valen")~ THEN BEGIN eat_valen
    SAY @1090
    IF ~~ THEN DO
~Enemy() 
ReallyForceSpell("Valen",CLERIC_HOLY_SMITE)
ReallyForceSpell("Valen",CLERIC_HOLY_WORD)
ReallyForceSpell("Valen",CLERIC_SUMMON_DEVA)
ReallyForceSpell(Myself,CLERIC_NEGATIVE_PLANE_PROTECTION)
ReallyForceSpell(Myself,CLERIC_IRONSKIN)
ReallyForceSpell(Myself,CLERIC_FALSE_DAWN)
ReallyForceSpell(Myself,CLERIC_SUNRAY)
ReallyForceSpell(Myself,CLERIC_NATURE_BEAUTY)
ReallyForceSpell(Player1,CLERIC_IMPLOSION)
ReallyForceSpell(Player2,CLERIC_IMPLOSION)
ReallyForceSpell(Player3,CLERIC_IMPLOSION)
ReallyForceSpell(Player4,CLERIC_IMPLOSION)
ReallyForceSpell(Player5,CLERIC_IMPLOSION)
ReallyForceSpell(Player6,CLERIC_IMPLOSION)~
  EXIT END
END

INTERJECT SARVAM01 0 Valen_TOB8
  == "VALENJ"   IF ~InParty("Valen")~ THEN @1100 = @1101
END SARVAM01 2

INTERJECT_COPY_TRANS GROMNIR 7 Valen_TOB9
  == "VALENJ"   IF ~InParty("Valen")~ THEN @1110
  == "GROMNIR"  IF ~InParty("Valen")~ THEN #40665
END 


valen/languages/valenint.d
////////////////////////////////////////////////////////////////////////////
// Valen SoA Interjections
////////////////////////////////////////////////////////////////////////////

INTERJECT RIBALD 0 Valen_IJ0
  == "RIBALD" IF ~InParty("Valen")~ THEN @1001
  == "VALENJ" IF ~InParty("Valen")~ THEN @1002
  == "RIBALD" IF ~InParty("Valen")~ THEN @1003
  == "VALENJ" IF ~InParty("Valen")~ THEN @1004
END RIBALD 46

EXTEND_BOTTOM KAMIR 2
  IF ~InParty("Valen")~ THEN EXTERN "VALENJ" hate_kamir
END

CHAIN VALENJ hate_kamir
   @1010
== "KAMIR" @1011 = @1012
== "VALENJ" @1013
END KAMIR bless_me

APPEND KAMIR
  IF ~~ THEN BEGIN bless_me
    SAY @1014
    IF ~~ THEN DO
 ~Enemy()
  ReallyForceSpell(Myself,CLERIC_NEGATIVE_PLANE_PROTECTION)
  ReallyForceSpell(Myself,CLERIC_HOLY_POWER)
  ReallyForceSpell(Myself,CLERIC_DRAW_UPON_HOLY_MIGHT)
  ReallyForceSpell(Myself,CLERIC_TRUE_SIGHT)
  ReallyForceSpell(Myself,CLERIC_CHAOTIC_COMMANDS)
  ReallyForceSpell(Myself,CLERIC_DEATH_WARD)
  ReallyForceSpell("Valen",CLERIC_HOLY_SMITE)
  ReallyForceSpell(Myself,WIZARD_BLACK_BLADE_OF_DISASTER)
  ReallyForceSpell(Myself,WIZARD_IMPROVED_HASTE)
  ReallyForceSpell(Myself,WIZARD_TENSERS_TRANSFORMATION)
  ReallyForceSpell(Myself,WIZARD_SPELL_IMMUNITY_ABJURATION)
  ChangeAIScript("wtasight",OVERRIDE)
  Attack("Valen")~ EXIT
  END
END 

// Antenos: bodhi servant
INTERJECT ANTENOS 0 Valen_IJ1
  == "VALENJ" IF ~InParty("Valen")~ THEN @1020
  == "ANTENOS" IF ~InParty("Valen")~ THEN @1021
END ANTENOS 1

// Chore: bodhi servant
INTERJECT CHORE 0 Valen_IJ2
  == "VALENJ" IF ~InParty("Valen")~ THEN @1022
END CHORE 1

// Fetch: bodhi servant
INTERJECT FETCH 0 Valen_IJ3
  == "VALENJ" IF ~InParty("Valen")~ THEN @1023
END FETCH 1

// Food: bodhi servant
INTERJECT FOOD 0 Valen_IJ4
  == "VALENJ" IF ~InParty("Valen")~ THEN @1024
END FOOD 1

// Master Cohn: bodhi servant
INTERJECT COHNTA 1 Valen_IJ5
  == "VALENJ" IF ~InParty("Valen")~ THEN @1025
  == "COHNTA" IF ~InParty("Valen")~ THEN @1026
  == "VALENJ" IF ~InParty("Valen")~ THEN @1027
END COHNTA 4

// Meredath: bodhi servant
INTERJECT LAUNE 3 Valen_IJ6
  == "VALENJ" IF ~InParty("Valen")~ THEN @1028 = @1029
END LAUNE 2

// Logan Coprith
INTERJECT CELOGAN 11 Valen_IJ7
  == "VALENJ" IF ~InParty("Valen")~ THEN #7722
END CELOGAN 17

// Khellon 
INTERJECT TREVIL01 6 Valen_IJ8
  == "VALENJ" IF ~InParty("Valen")~ THEN #50502
END TREVIL01 10

// Pauden, druid grove
INTERJECT CEDRUID1 6 Valen_IJ9
  == "VALENJ" IF ~InParty("Valen")~ THEN #9272
END CEDRUID1 12
INTERJECT CEDRUID1 12 Valen_IJ10
  == "VALENJ" IF ~InParty("Valen")~ THEN #9319
END CEDRUID1 13

// Samia
INTERJECT SAMIA 26 Valen_IJ11
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1030
  == "SAMIA"     IF ~InParty("Valen")~ THEN @1031
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1032
  == "SAMIA"     IF ~InParty("Valen")~ THEN @1033
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1034
END SAMIA 18

// Grancor
INTERJECT FIRWLF01 0 Valen_IJ12
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1040
  == "FIRWLF01"     IF ~InParty("Valen")~ THEN @1041
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1042 = @1043
END FIRWLF01 7

// Firkraag
INTERJECT FIRKRA02 15 Valen_IJ13
  == "VALENJ"     IF ~InParty("Valen")~ THEN #49290
END FIRKRA02 28

// Daleson
EXTEND_BOTTOM DALESON 9 10 11
  IF ~InParty("Valen")~ THEN EXTERN VALENJ eat_dale
END

EXTEND_BOTTOM DELCIA 10 16 30 31 32 33
  IF ~InParty("Valen")~ THEN EXTERN VALENJ eat_del
END

APPEND VALENJ
  IF ~~ THEN BEGIN eat_dale
    SAY @1044
    IF ~~ THEN DO ~ActionOverride("Daleson",Enemy())
                   DisplayString("Daleson",19778)
                   Attack("Daleson")~ EXIT
  END
  IF ~~ THEN BEGIN eat_del
    SAY @1045 = @1046
    IF ~~ THEN DO ~ActionOverride("Delcia",Enemy())
                   DisplayString("Delcia",19751)
                   ActionOverride("kpsold01",SetNumTimesTalkedTo(1))
                   ActionOverride("kpsold01",Enemy())
                   Attack("Delcia")~ EXIT
  END
END

// Amnish Guards!
INTERJECT VVAMN3 0 Valen_IJ14
  == "VVAMN3"     IF ~InParty("Valen")~ THEN @1050
  == "VVAMN1"     IF ~InParty("Valen")~ THEN @1051
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1052
  == "VVAMN3"     IF ~InParty("Valen")~ THEN @1053
END VVAMN1 fight_guards

APPEND VVAMN1
  IF ~~ THEN BEGIN fight_guards
    SAY @1053
    IF ~~ THEN DO 
~ActionOverride("vvamn2",Enemy())
 ActionOverride("vvamn2",Attack("Valen"))
 ActionOverride("vvamn3",Enemy())
 ActionOverride("vvamn3",Attack("Valen"))
 ActionOverride("Valen",ReallyForceSpell("Valen",4152))
 ActionOverride("Valen",Attack("vvamn1"))
 Enemy()
 Attack("Valen")~ EXIT
  END
END

// Delon
EXTEND_BOTTOM DELON 1 5 9 15 16 
  IF ~InParty("Valen")~ THEN EXTERN VALENJ eat_delon1
END

CHAIN VALENJ eat_delon1
  @1060 = @1061
== "DELON" @1062
END VALENJ eat_delon2

APPEND VALENJ
  IF ~~ THEN BEGIN eat_delon2
    SAY @1063
    IF ~~ THEN DO ~ActionOverride("Delon",Enemy())
                   DisplayString("Delon",19778)
                   Attack("Delon")~ EXIT
  END
END

// Chief inspector Brega
EXTEND_BOTTOM HABREGA 1
  IF ~InParty("Valen")~ THEN EXTERN HABREGA eat_brega1
END

CHAIN HABREGA eat_brega1
   @1070 
== "VALENJ" @1071
== "HABREGA" @1072
== "VALENJ" @1073
== "HABREGA" @1074
== "VALENJ" @1075
END HABREGA eat_brega2

APPEND HABREGA
  IF ~~ THEN BEGIN eat_brega2
    SAY @1076
    IF ~~ THEN DO ~ActionOverride("Valen",Attack("habrega"))
                   Shout(99)
                   Enemy() 
                   Attack("Valen")~ EXIT
  END
END

// Rose
APPEND MURDGIRL
  IF WEIGHT #-1 ~Name("Valen",LastTalkedToBy)
                 Global("ValenRose","LOCALS",0)~ THEN BEGIN rose1
    SAY @1080
    IF ~~ THEN DO ~SetGlobal("ValenRose","LOCALS",1)~ EXTERN VALENJ rose2
  END
  IF ~~ THEN BEGIN rose3
    SAY @1084
    IF ~~ THEN EXIT
  END
END

CHAIN VALENJ rose2
  @1081
== "MURDGIRL" @1082
== "VALENJ" @1083
END MURDGIRL rose3

// Valygar
INTERJECT VALYGAR 20 Valen_IJ15
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1090
  == "VALYGAR"     IF ~InParty("Valen")~ THEN @1091
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1092
  == "VALYGAR"     IF ~InParty("Valen")~ THEN @1093
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1094
END VALYGAR 27

// Anath
INTERJECT RNGWLF01 11 Valen_IJ16
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1100
  == "RNGWLF01"     IF ~InParty("Valen")~ THEN @1101
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1102
END RNGWLF01 7

// Roger
APPEND ROGER
  IF WEIGHT #-1 ~InParty("Valen")
      Global("ValenRoger","GLOBAL",0)~ THEN BEGIN vr0
    SAY @1110
    IF ~~ THEN DO ~SetGlobal("ValenRoger","GLOBAL",1)~ EXTERN VALENJ valen_rog
  END
  CHAIN2 VALENJ valen_rog
    @1111
  == @1112
  == @1113
  == @1114
  == @1115
  == @1116
  == @1117
  == @1118
  == @1119
  END ROGER valen_bye
  IF ~~ THEN BEGIN valen_bye
    SAY @1120
    IF ~~ THEN EXIT
  END
END

INTERJECT ARAN02 0 Valen_IJ17
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1130
  == "ARAN02"     IF ~InParty("Valen")~ THEN @1131
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1132
END ARAN02 aran_bye

APPEND ARAN02
  IF ~~ THEN BEGIN aran_bye
    SAY #61774
    IF ~~ THEN DO 
      ~Enemy()
       ReallyForceSpell(Myself,CLERIC_FALSE_DAWN)
       ReallyForceSpell(Myself,CLERIC_SUNRAY)~ EXIT
  END
END

INTERJECT PPIMOEN 0 Valen_IJ18
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1140
  == "PPIMOEN"     IF ~InParty("Valen")~ THEN @1141
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1142
END PPIMOEN 1

// Lonk
INTERJECT PPWORKER 17 Valen_IJ19
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1150
  == "PPWORKER"   IF ~InParty("Valen")~ THEN @1151
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1152
END PPWORKER 21

// Dace
INTERJECT PPJOYE 0 Valen_IJ20
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1160
  == "PPJOYE"     IF ~InParty("Valen")~ THEN @1161
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1162
  == "PPJOYE"     IF ~InParty("Valen")~ THEN @1163
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1164
  == "PPJOYE"     IF ~InParty("Valen")~ THEN @1165 = @1166 = @1167
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1168
  == "PPJOYE"     IF ~InParty("Valen")~ THEN @1169 
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1170
END PPJOYE die_now

APPEND PPJOYE
  IF ~~ THEN BEGIN die_now
    SAY @1171
    IF ~~ THEN DO ~Enemy()
      ReallyForceSpell(Myself,WIZARD_PROTECTION_FROM_MAGIC_WEAPONS)
      ReallyForceSpell(Myself,WIZARD_IMPROVED_HASTE)~ 
      JOURNAL #22934 EXIT
  END
END

INTERJECT PPSAEM2 21 Valen_IJ21
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1180
  == "PPSAEM2"     IF ~InParty("Valen")~ THEN @1181
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1182
  == "PPSAEM2"     IF ~InParty("Valen")~ THEN @1183
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1184
  == "PPSAEM2"     IF ~InParty("Valen")~ THEN @1185
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1186
  == "PPSAEM2"     IF ~InParty("Valen")~ THEN @1187
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1188
  == "PPSAEM2"     IF ~InParty("Valen")~ THEN @1189
END PPSAEM2 20

// Svirfneblin
INTERJECT UDSVIR08 3 Valen_IJ22
  == "VALENJ"     IF ~InParty("Valen")~ THEN #51564
END UDSVIR08 1

// Demon Lord
INTERJECT UDDEMON 20 Valen_IJ23
  == "VALENJ"     IF ~InParty("Valen")
                      Dead("Phaere")
                      PartyHasItem("MISC9t")~ THEN @1190
END UDDEMON valen_trade

APPEND UDDEMON
  IF ~~ THEN BEGIN valen_trade
    SAY @1191
    IF ~~ THEN DO  ~TakePartyItem("MISC9t")
SetGlobal("EggsDie","GLOBAL",1)
GiveItemCreate("HALB06","Valen",1,1,1)
GiveItemCreate("RING07","Valen",1,1,1)
AddXPObject("Valen",500000)~ GOTO 11
// normally 150000 is distributed to the party
  END
END


INTERJECT UDSIMYAZ 36 Valen_IJ24
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1200
  == "UDSIMYAZ"     IF ~InParty("Valen")~ THEN #52229
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1201
  == "UDSIMYAZ"     IF ~InParty("Valen")~ THEN @1202
END VALENJ eat_sim

APPEND VALENJ
  IF ~~ THEN BEGIN eat_sim
    SAY @1203
    IF ~~ THEN DO ~EraseJournalEntry(57863)
  SetGlobal("AttackedGith","GLOBAL",1)
  ActionOverride("UDSIMYAZ",Enemy())~ EXIT
  END
END

EXTEND_BOTTOM c6drizz1 25 29 30 34 36 54
  IF ~InParty("Valen")~ THEN EXTERN VALENJ eat_driz
END

CHAIN VALENJ eat_driz
  @1210
== "c6drizz1" @1211
== "valenj" @1212
== "c6drizz1" @1213
== "valenj" @1214
END c6drizz1 eat_driz2

APPEND c6drizz1
  IF ~~ THEN BEGIN eat_driz2
    SAY @1215
    IF ~~ THEN DO ~Shout(89)
                   Enemy()
                   CreateCreatureObjectEffect("C6GUEN","SPANISUM",Myself)~ EXIT
  END
END

// Elhan
INTERJECT c6elhan2 67 Valen_IJ25
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1220
END c6elhan2 68

// Demin
INTERJECT sudemin 26 Valen_IJ26
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1230
END sudemin 27

// Wraith Sarevok
INTERJECT hellsare 25 Valen_IJ27
  == "VALENJ"     IF ~InParty("Valen")~ THEN @1240
END hellsare 26


valen/valen.d
EXTEND_BOTTOM BOANASTE 0
  IF ~Global("WhoAreYou","LOCALS",0)~ THEN REPLY @1 
    DO ~SetGlobal("WhoAreYou","LOCALS",1)~ GOTO whoareyou
  IF ~Global("YouArePale","LOCALS",0)
      CheckStatGT(Player1,15,INT)~ THEN REPLY @9 
    DO ~SetGlobal("YouArePale","LOCALS",1)~ GOTO youarepale
END

EXTEND_BOTTOM BOANASTE 5
  IF ~CheckStatGT(Player1,15,INT)~ THEN REPLY @4 
    GOTO vampire
END

APPEND BOANASTE
  IF ~~ THEN BEGIN whoareyou
    SAY @2
    COPY_TRANS BOANASTE 0
  END
  IF ~~ THEN BEGIN youarepale
    SAY @3
    COPY_TRANS BOANASTE 0
  END
  IF ~~ THEN BEGIN vampire
    SAY @5 = @6 = @7 = @8
    IF ~~ THEN GOTO 8 
  END
END

/////////////////////////////////////////////////////////////////////
REPLACE BODHI 
  IF ~~ THEN BEGIN 122
    SAY #49062 
    IF ~~ THEN DO ~CreateCreatureObject("valen",Myself,1,1,1)~ GOTO bodhi_valen
  END

  IF ~Global("BodhiAppear","GLOBAL",1)
      !Global("WorkingForAran","GLOBAL",1)
      Global("BodhiMove","LOCALS",0)
      PartyGoldGT(14999)~ 
      // removed: !Class(Player1,THIEF)
  THEN BEGIN 0 
    SAY #21339 
    IF ~~ THEN REPLY #21478 GOTO 1
    IF ~~ THEN REPLY #21479 GOTO 1
    IF ~~ THEN REPLY #21480 GOTO 1
    IF ~~ THEN REPLY #21508 GOTO 1
  END
END

APPEND BODHI
  IF ~~ THEN BEGIN bodhi_valen
    SAY @20 = @21 = @22 = @23 = @24
    COPY_TRANS BODHI 122
  END
END

EXTEND_BOTTOM PPBODHI4 15
  IF     ~InParty("Valen")
          Global("DoValenPPOnce","LOCALS",0)~ THEN 
   DO ~SetGlobal("DoValenPPOnce","LOCALS",1)~ EXTERN VALENJ ppbodhi
END

EXTEND_BOTTOM C6BODHI 22
  IF ~InParty("Valen")~ THEN EXTERN VALENJ c6bodhi
END


EXTEND_BOTTOM PLAYER1 33 // tree of life gut check 
  IF ~InParty("Valen")~ 
    THEN EXTERN PLAYER1 tree_of_life
END

APPEND PLAYER1 
  IF ~~ THEN BEGIN tree_of_life
    SAY @400
    IF ~~ THEN REPLY @401 EXTERN VALENJ tol_1
    IF ~~ THEN REPLY @402 EXTERN VALENJ tol_2
  END
END


/////////////////////////////////////////////////////////////////////

BEGIN VALEN

IF ~True()~ THEN BEGIN first
  SAY @100
  IF ~Global("Who","LOCALS",0)~ THEN 
    DO ~SetGlobal("Who","LOCALS",1)~ REPLY @101 GOTO who
  IF ~Global("What","LOCALS",0)~ THEN 
    DO ~SetGlobal("What","LOCALS",1)~ REPLY @121 GOTO what
  IF ~~ THEN REPLY @131 EXIT
  IF ~~ THEN REPLY @141 GOTO join
END


IF ~~ THEN BEGIN who
  SAY @102 = @103 = @104 = @105 = @106 = @107 = @108 = @109 = @110 =
      @111 = @112 = @113 = @114
  IF ~~ THEN GOTO first
END

IF ~~ THEN BEGIN what
  SAY @122 = @123 = @124 = @125 = @126 = @127
  IF ~~ THEN GOTO first
END

IF ~~ THEN BEGIN join
  SAY @142 
  IF ~~ THEN DO ~JoinParty()~ EXIT
END

// kicked-out dialogue

BEGIN VALENP

IF ~True()~ THEN BEGIN first
  SAY @200
  IF ~~ THEN REPLY @201 DO ~JoinParty()~ EXIT
  IF ~~ THEN REPLY @202 EXIT
END

// join dialogue

BEGIN VALENJ

IF ~InParty("Aerie")~ THEN BEGIN bad1
  SAY @301 IF ~~ THEN DO ~SetLeavePartyDialogFile() LeaveParty()~ EXIT
END

IF ~InParty("Anomen")
    OR(2)
     Alignment("Anomen",LAWFUL_GOOD)
     Alignment("Anomen",LAWFUL_NEUTRAL)~ THEN BEGIN bad2
  SAY @302 IF ~~ THEN DO ~SetLeavePartyDialogFile() LeaveParty()~ EXIT
END

IF ~InParty("Keldorn")~ THEN BEGIN bad3
  SAY @300 IF ~~ THEN DO ~SetLeavePartyDialogFile() LeaveParty()~ EXIT
END

IF ~InParty("Mazzy")~ THEN BEGIN bad4
  SAY @303 IF ~~ THEN DO ~SetLeavePartyDialogFile() LeaveParty()~ EXIT
END

IF ~!InParty(Myself)~ THEN BEGIN not_here
  SAY @200
  IF ~~ THEN REPLY @201 DO ~JoinParty()~ EXIT
  IF ~~ THEN REPLY @202 DO ~SetLeavePartyDialogFile()~ EXIT
END

IF ~True()~ THEN BEGIN chat
  SAY @600
  IF ~Global("ValenNoHide","LOCALS",0)~ THEN REPLY @601 
    DO ~SetGlobal("ValenNoHide","LOCALS",1)~ GOTO very_well
  IF ~Global("ValenNoHide","LOCALS",1)~ THEN REPLY @602
    DO ~SetGlobal("ValenNoHide","LOCALS",0)~ GOTO very_well
  IF ~Global("ValenNoCharm","LOCALS",0)~ THEN REPLY @607 
    DO ~SetGlobal("ValenNoCharm","LOCALS",1)~ GOTO very_well
  IF ~Global("ValenNoAttack","LOCALS",0)~ THEN REPLY @603
    DO ~SetGlobal("ValenNoAttack","LOCALS",1)~ GOTO dont_like
  IF ~Global("ValenNoAttack","LOCALS",1)~ THEN REPLY @604
    DO ~SetGlobal("ValenNoAttack","LOCALS",0)~ GOTO oh_yes
  IF ~Global("ValenNoCharm","LOCALS",1)~ THEN REPLY @608 
    DO ~SetGlobal("ValenNoCharm","LOCALS",0)~ GOTO very_well
  IF ~Global("ValenNoDetect","LOCALS",0)~ THEN REPLY @609
    DO ~SetGlobal("ValenNoDetect","LOCALS",1)~ GOTO very_well
  IF ~Global("ValenNoDetect","LOCALS",1)~ THEN REPLY @610
    DO ~SetGlobal("ValenNoDetect","LOCALS",0)~ GOTO very_well
  IF ~~ THEN REPLY @605 
    DO ~SetGlobal("ValenFixUp","LOCALS",1)~ GOTO very_well
  IF ~~ THEN REPLY @606 GOTO very_well
END

IF ~~ THEN BEGIN very_well SAY @699 IF ~~ THEN EXIT END
IF ~~ THEN BEGIN dont_like SAY @698 IF ~~ THEN EXIT END
IF ~~ THEN BEGIN oh_yes SAY @697 IF ~~ THEN EXIT END

IF ~~ THEN BEGIN tol_1
  SAY @403 = @404
  COPY_TRANS PLAYER1 33
END
IF ~~ THEN BEGIN tol_2
  SAY @405 = @406 = @407 
  COPY_TRANS PLAYER1 33
END

IF ~~ THEN BEGIN ppbodhi
  SAY @310
  IF ~~ THEN EXTERN PPBODHI4 valen0
END

CHAIN2 PPBODHI4 valen0
   @311 = @312 = @313 
== @314 
== @315 
== @316 
== @317 = @318 
== @319
== @320 
== @321 END PPBODHI4 valen1

APPENDI PPBODHI4
  IF ~~ THEN BEGIN valen1
    SAY @322
    COPY_TRANS PPBODHI4 15
  END
END

IF ~~ THEN BEGIN c6bodhi
  SAY @330
  IF ~~ THEN EXTERN C6BODHI valen2
END

CHAIN2 C6BODHI valen2
   @331 = @332
== @333
== @334 = @335
== @336
END C6BODHI valen3

APPENDI C6BODHI
  IF ~~ THEN BEGIN valen3
    SAY @337
    COPY_TRANS C6BODHI 22
  END
END

// salvanas

APPEND SALVANAS

IF WEIGHT #-1 ~!InPartySlot(LastTalkedToBy,0)
               Name("Valen",LastTalkedToBy)~ THEN BEGIN s0 
  SAY @500
  IF ~~ THEN EXTERN VALENJ s1
END

CHAIN2 VALENJ s1
   @501 
== @502
== @503
END SALVANAS s2

IF ~~ THEN BEGIN s2
  SAY @504
  IF ~~ THEN DO ~EscapeArea()~ EXIT
END

END
